### ISSUE TITLE
Update `coordinateWithBearing` to use Google's API

### ISSUE THREAD

**Issue by furzik** [ProjectDent/ARKit-CoreLocation#239]
I've found this extension to be much more accurate than what you currently have here:
https://github.com/ProjectDent/ARKit-CoreLocation/blob/f22f51bfc6b98d5b2ef54737ae242b40ae6f087e/Sources/ARKit-CoreLocation/Extensions/CLLocation%2BExtensions.swift#L107
```
extension CLLocationCoordinate2D {
func coordinateWithBearing(bearing: CLLocationDegrees, distanceMeters: CLLocationDistance) -> CLLocationCoordinate2D {
        return GMSGeometryOffset(self, distance, heading)
    }
}
```

**Comment by aaronbrethorst** [comment]
There are significant costs associated with using Google's Maps SDK, both in monetary and device footprint terms. This would be a major dependency for this project to add for the sake of replacing `coordinateWithBearing`.

**Comment by Pilot-Marc** [comment]
I use `coordinateWithBearing` extensively and have seen no problems.  Can you elaborate on why you don't think it's as robust?

**Comment by halmueller** [comment]
I concur with Scott that coordinateWithBearing is not particularly accurate.

See the unit tests I added in CLLocationExtensionsTests.swift. If you check the Test scheme, you’ll see that they’re all disabled. That’s because they were causing CI to fail. The underlying formula uses a rectangular approximation of a sphere, not a geodesic (great circle), which would be correct. I have a private branch open to fix the formula and get those tests passing, but I haven’t had any time to devote to that bit lately.

The unit test file contains some sample inputs/outputs generated with PostGIS, and compares those known good values to coordinateWithBearing.

**Comment by Pilot-Marc** [comment]
I stand corrected.  My code uses this extension which compensates for the earth's curvature...

```
public extension CLLocationCoordinate2D {

    func coordinateWithBearing(bearing: Double, distanceMeters: Double) -> CLLocationCoordinate2D {

        let distRadiansLat = distanceMeters.metersToLatitude // earth radius in meters latitude
        let distRadiansLong = distanceMeters.metersToLongitude // earth radius in meters longitude

        let lat1 = self.latitude * Double.pi / 180
        let lon1 = self.longitude * Double.pi / 180

        let lat2 = asin(sin(lat1) * cos(distRadiansLat) + cos(lat1) * sin(distRadiansLat) * cos(bearing))
        let lon2 = lon1 + atan2(sin(bearing) * sin(distRadiansLong) * cos(lat1), cos(distRadiansLong) - sin(lat1) * sin(lat2))

        return CLLocationCoordinate2D(latitude: lat2 * 180 / Double.pi, longitude: lon2 * 180 / Double.pi)
    }
}

```

**Comment by halmueller** [comment]
That's the code that's in the develop branch as well, and that's what I've written the failing tests against. I noted in the docstring that it's probably good enough for short distances, but I find errors of a few degrees and a few tens of meters at a distanceToMeters: value of 500. 

For a heading sensor with a precision of about 15 degrees, and fairly short (a few km) radiuses, what's in there is probably good enough, though.

**Comment by halmueller** [comment]
Mostly resolved by PR #244. Remaining errors carried over to Issue #246.
